### PR TITLE
Update toolchain to nightly 2023-08-04

### PR DESCRIPTION
**Pull request by remi-delmas-3000**
### Description of changes: 

bump toolchain to nightly-2023-08-04

### Call-outs:

- disables `tests/expected/slice_c_str/c_str.c` because of ongoing refactor of c_str_literals https://github.com/rust-lang/rust/issues/105723

### Testing:

CI tests are passing. Built and tested on Apple M1.

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [ ] Methods or procedures are documented
- [ ] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

**Review by celinval — COMMENTED** [review]
It looks good. I was just wondering if there's any specific reason why you updated our dependencies. If there's no reason, can you please revert those changes and push a separate PR?

BTW, thanks for fixing the documentation warnings. We should probably enforce no warnings in our CI but not here.

Thanks again!

File: Cargo.lock
[the remark this answers is not in the document]
Comment (remi-delmas-3000) [reply Cargo.lock:1]: that's an accident indeed

**Comment by remi-delmas-3000** [comment]
> It looks good. I was just wondering if there's any specific reason why you updated our dependencies. If there's no reason, can you please revert those changes and push a separate PR?
> 
> BTW, thanks for fixing the documentation warnings. We should probably enforce no warnings in our CI but not here.
> 
> Thanks again!

there's a clang-format-14 reports formatting error on the vec.c stub, but I could not reproduce it locally so I could not fix.

**Review by adpaco-aws — APPROVED** [review]
Thanks @remi-delmas-3000 !

> here's a clang-format-14 reports formatting error on the vec.c stub, but I could not reproduce it locally so I could not fix.

Looks like it's been there for a while (seen it in other PRs) and our workflow is set to continue on errors from clang-format.

**Pull request merged**